v3/src/components/web-view/web-view-utils.ts

```diff
@@ -0,0 +1,23 @@
+export const kRootPluginUrl = "https://codap-resources.s3.amazonaws.com/plugins"
+export const kRelativePluginRoot = "../../../../extn/plugins"
+
+export function processPluginUrl(url: string) {
+  let updatedUrl = url
+
+  // Some plugins relied on index.html being the default file loaded when pointing to a directory.
+  // This is no longer the case with our S3 hosted plugins, so we have to modify the path to point to the
+  // correct file in this case.
+  if (updatedUrl.endsWith("/")) updatedUrl = `${updatedUrl}index.html`
+
+  // Many plugins used to be packaged with CODAP v2, then referenced relatively.
+  // These references need to be changed to point to the S3 bucket we now use to host plugins.
+  if (updatedUrl.startsWith(kRelativePluginRoot)) {
+    updatedUrl = `${kRootPluginUrl}${updatedUrl.slice(kRelativePluginRoot.length)}`
```

Comment:
Codecov reports that this line is not covered with a test. `processPluginUrl` is a perfect candidate for a jest test which tests/demonstrates how various url combinations are handled.